src/lib.rs

```diff
@@ -30,1209 +106,244 @@ extern crate core;
 
 #[cfg(all(feature = "alloc", not(feature = "std"), not(test)))]
 use alloc::{string::String, vec::Vec};
-use core::convert::{Infallible, TryFrom};
-use core::{fmt, mem};
+use core::fmt;
 
+#[doc(inline)]
 pub use crate::primitives::checksum::Checksum;
-use crate::primitives::checksum::{self, PackedFe32};
+#[cfg(feature = "alloc")]
+use crate::primitives::decode::{
+    CheckedHrpstring, CheckedHrpstringError, ChecksumError, UncheckedHrpstring,
+};
+#[doc(inline)]
 pub use crate::primitives::gf32::Fe32;
-use crate::primitives::hrp;
+#[doc(inline)]
 pub use crate::primitives::hrp::Hrp;
+#[doc(inline)]
 pub use crate::primitives::iter::{ByteIterExt, Fe32IterExt};
-pub use crate::primitives::{Bech32, Bech32m};
+#[doc(inline)]
+pub use crate::primitives::{Bech32, Bech32m, NoChecksum};
 
 mod error;
+/// Re-exports the hrp types from [`primitives::hrp`] to make importing ergonomic for the top level APIs.
+pub mod hrp;
+/// All the primitive types and functionality used in encoding and decoding.
 pub mod primitives;
+/// API for encoding and decoding segwit addresses.
```

Comment:
In the docs, this comment collides with the existing module-level docs in `segwit.rs`, giving us:

> API for encoding and decoding segwit addresses. Segregated Witness API - enables typical usage for encoding and decoding segwit addresses.

We should remove one of the comments to avoid the redundant sentence.

Reply:
Ha, I saw this once and didn't fix it. Thanks for taking the time to build the docs and check them!

I've removed the docs from `rust-bech32/src/hrp.rs` and clippy seems to be ok with it, although I'm unsure why. Like @apoelstra said above we saw a new lint trigger for this situation in rust-bitcoin, will see how it goes in CI.